packages/astro/test/csrf-protection.test.js

```diff
@@ -176,6 +176,34 @@ describe('CSRF origin check', () => {
 		});
 	});
 
+	it("return a 200 when the origin doesn't match but calling HEAD", async () => {
```

Comment:
Since this is quite a delicate feature that involves security, would you mind adding a test for each verb method that was added to `SAFE_METHODS`? I would like to cover all cases if possible